File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] ret = new int[2];
        int[] rank = {6, 5, 4, 3, 2, 1, 0};
        int[] cnt = new int[46];
        int blank = 0;
        int mn = 0;
        for(int i : win_nums) cnt[i]++;
        for(int i : lottos){
           if(i == 0) {blank++; continue;} 
            for(int j = 1; j <= 45; j++){
                if(cnt[j] == 1 && i == j){
                    mn++;
                }
            }
        }
        int mx = mn + blank;
        for(int i = 0; i < rank.length; i++){
            if(mx == rank[i]) ret[0] = (i+1)>= 6 ? 6 : (i+1);
            if(mn == rank[i]) ret[1] = (i+1)>= 6 ? 6 : (i+1);
            if(ret[0] != 0 && ret[1] != 0) break;
        }
        return ret;
    }
}


/*
1.당첨 가능했던 최고, 최저
2. 순서상관없이 숫자 있으면 맞힌것
3.
*/

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
from collections import Counter

def solution(lottos, win_nums):
    rank = {6:1, 5:2, 4:3, 3:4, 2:5, 1:6, 0:6}
    cnt = Counter(lottos)
    mmn = len(set(lottos) & set(win_nums))
    mmx = mmn + cnt[0]
    return [rank[mmx], rank[mmn]]
    
        
    
            
    
"""
일부번호 인식불가,
최고, 최저순위 파악
다맞을 때와 다틀릴때
[최고, 최저] 반환
1. 같은 숫자 갯수 구하기
2. 0인 갯수 구하기
3. 처음 같은 숫자 갯수는 최저, 0인 갯수 더한게 최고

"""